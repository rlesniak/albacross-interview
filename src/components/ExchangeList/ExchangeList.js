// @flow

import * as React from 'react';
import type { Map } from 'immutable';

import CurrencyItem from 'Components/CurrencyItem';
import { getMainCurrency, currencySignMap } from 'Utils/helpers';

type PropsType = {
  currentCurrency: string,
  exchangeList: Map<string, number>,
};

const addSign = (value, currency) =>
  `${value.toFixed(2)}${currencySignMap[currency]}`;

const ExchangeList = ({
  exchangeList,
  currentCurrency,
}: PropsType) => (
  <div className="currency-list">
    <h3>1 {currentCurrency} is equal:</h3>
    {exchangeList
      .entrySeq()
      .map((value, i) => (
        <CurrencyItem
          key={i}
          value={addSign(value[1], currentCurrency)}
          name={getMainCurrency(value[0])}
        />
      ))}
  </div>
);

export default ExchangeList;
