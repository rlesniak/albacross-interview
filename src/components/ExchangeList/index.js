// @flow

export { default } from './ExchangeList';
