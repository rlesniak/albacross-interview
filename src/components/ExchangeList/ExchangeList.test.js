import React from 'react';
import { shallow } from 'enzyme';
import { Map } from 'immutable';

import ExchangeList from './ExchangeList';

describe('<ExchangeList />', () => {
  const list = Map({ USD: 1.22, PLN: 0.43, BTC: 0.00001 });
  it('should render without crash', () => {
    const component = shallow(<ExchangeList currentCurrency="USD" exchangeList={list} />);

    expect(component).toMatchSnapshot();
  });

  it('should render different currency', () => {
    const component = shallow(<ExchangeList currentCurrency="PLN" exchangeList={list} />);

    expect(component).toMatchSnapshot();
  });
});

