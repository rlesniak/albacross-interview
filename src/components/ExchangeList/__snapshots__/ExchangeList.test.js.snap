// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExchangeList /> should render different currency 1`] = `
<div
  className="currency-list"
>
  <h3>
    1 
    PLN
     is equal:
  </h3>
  <withHandlers(CurrencyItem)
    key="0"
    name="USD"
    value="1.22zł"
  />
  <withHandlers(CurrencyItem)
    key="1"
    name="PLN"
    value="0.43zł"
  />
  <withHandlers(CurrencyItem)
    key="2"
    name="BTC"
    value="0.00zł"
  />
</div>
`;

exports[`<ExchangeList /> should render without crash 1`] = `
<div
  className="currency-list"
>
  <h3>
    1 
    USD
     is equal:
  </h3>
  <withHandlers(CurrencyItem)
    key="0"
    name="USD"
    value="1.22$"
  />
  <withHandlers(CurrencyItem)
    key="1"
    name="PLN"
    value="0.43$"
  />
  <withHandlers(CurrencyItem)
    key="2"
    name="BTC"
    value="0.00$"
  />
</div>
`;
